assignment-1/pkg/controller/backupController.go

```diff
@@ -127,29 +131,48 @@ func (bc *BackupController) createSnapshot(bgContext context.Context,
 			},
 		},
 	}
-	_, err = bc.Clients.ExternalSnapshotClientSet.SnapshotV1().VolumeSnapshots(
+	volumeSnapshot, err := bc.Clients.ExternalSnapshotClientSet.SnapshotV1().VolumeSnapshots(
 		pvc.Namespace).Create(bgContext, snapTemplate, metav1.CreateOptions{})
 	if err != nil {
-		bc.updateStatus(bgContext, FAILED, snapshot)
+		bc.updateStatus(bgContext, FAILED, snapshot.Namespace, snapshot.Name, "", "")
 		return err
 	}
-	bc.updateStatus(bgContext, COMPLETED, snapshot)
+	if snapshot.Spec.WaitForVolumeSnapshot {
+		bc.waitForVolumeSnapshot(bgContext, volumeSnapshot)
+	}
+	bc.updateStatus(bgContext, COMPLETED, snapshot.Namespace, snapshot.Name,
+		volumeSnapshot.Name, volumeSnapshot.Namespace)
 	fmt.Println("snapshot created")
 	return nil
 }
 
-func (bc *BackupController) updateStatus(bgContext context.Context, state string,
-	snapshot *v1alpha1.SnapshotBackup) error {
-	snapshotResource, err := bc.SnapshotLister.SnapshotBackups(snapshot.Namespace).Get(
-		snapshot.Name)
+func (bc *BackupController) waitForVolumeSnapshot(bgContext context.Context,
+	volumeSnapshot *volumesnapshotv1.VolumeSnapshot) {
+	state := *volumeSnapshot.Status.ReadyToUse
+	for !state {
+		vsResource, _ := bc.Clients.ExternalSnapshotClientSet.SnapshotV1().VolumeSnapshots(
+			volumeSnapshot.Namespace).Get(bgContext, volumeSnapshot.Name, metav1.GetOptions{})
+		if *vsResource.Status.ReadyToUse {
+			break
+		}
+		state = *vsResource.Status.ReadyToUse
+		time.Sleep(20 * time.Second)
+	}
```

Comment:
Instead of a `For` loop, we could have used [https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#Poll](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#Poll) here to check whether snapshot is in ready state or not.